// @teishi/bulma_theme/dist/bulma-theme.es.js
import ke, { createContext as mr, useReducer as yr, useContext as Er, useState as je } from "react";
var ee = { exports: {} }, W = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ce;
function gr() {
  if (Ce) return W;
  Ce = 1;
  var E = ke, d = Symbol.for("react.element"), C = Symbol.for("react.fragment"), b = Object.prototype.hasOwnProperty, T = E.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, x = { key: !0, ref: !0, __self: !0, __source: !0 };
  function _(S, v, j) {
    var h, u = {}, w = null, M = null;
    j !== void 0 && (w = "" + j), v.key !== void 0 && (w = "" + v.key), v.ref !== void 0 && (M = v.ref);
    for (h in v) b.call(v, h) && !x.hasOwnProperty(h) && (u[h] = v[h]);
    if (S && S.defaultProps) for (h in v = S.defaultProps, v) u[h] === void 0 && (u[h] = v[h]);
    return { $$typeof: d, type: S, key: w, ref: M, props: u, _owner: T.current };
  }
  return W.Fragment = C, W.jsx = _, W.jsxs = _, W;
}
var $ = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var we;
function br() {
  return we || (we = 1, process.env.NODE_ENV !== "production" && function() {
    var E = ke, d = Symbol.for("react.element"), C = Symbol.for("react.portal"), b = Symbol.for("react.fragment"), T = Symbol.for("react.strict_mode"), x = Symbol.for("react.profiler"), _ = Symbol.for("react.provider"), S = Symbol.for("react.context"), v = Symbol.for("react.forward_ref"), j = Symbol.for("react.suspense"), h = Symbol.for("react.suspense_list"), u = Symbol.for("react.memo"), w = Symbol.for("react.lazy"), M = Symbol.for("react.offscreen"), re = Symbol.iterator, De = "@@iterator";
    function Fe(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = re && e[re] || e[De];
      return typeof r == "function" ? r : null;
    }
    var k = E.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function p(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        Ie("error", e, t);
      }
    }
    function Ie(e, r, t) {
      {
        var n = k.ReactDebugCurrentFrame, o = n.getStackAddendum();
        o !== "" && (r += "%s", t = t.concat([o]));
        var s = t.map(function(i) {
          return String(i);
        });
        s.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, s);
      }
    }
    var Ne = !1, We = !1, $e = !1, Me = !1, Ye = !1, te;
    te = Symbol.for("react.module.reference");
    function Le(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === b || e === x || Ye || e === T || e === j || e === h || Me || e === M || Ne || We || $e || typeof e == "object" && e !== null && (e.$$typeof === w || e.$$typeof === u || e.$$typeof === _ || e.$$typeof === S || e.$$typeof === v || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === te || e.getModuleId !== void 0));
    }
    function Ve(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var o = r.displayName || r.name || "";
      return o !== "" ? t + "(" + o + ")" : t;
    }
    function ne(e) {
      return e.displayName || "Context";
    }
    function R(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && p("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case b:
          return "Fragment";
        case C:
          return "Portal";
        case x:
          return "Profiler";
        case T:
          return "StrictMode";
        case j:
          return "Suspense";
        case h:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case S:
            var r = e;
            return ne(r) + ".Consumer";
          case _:
            var t = e;
            return ne(t._context) + ".Provider";
          case v:
            return Ve(e, e.render, "ForwardRef");
          case u:
            var n = e.displayName || null;
            return n !== null ? n : R(e.type) || "Memo";
          case w: {
            var o = e, s = o._payload, i = o._init;
            try {
              return R(i(s));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var P = Object.assign, F = 0, ae, ie, oe, se, ue, le, ce;
    function fe() {
    }
    fe.__reactDisabledLog = !0;
    function Ue() {
      {
        if (F === 0) {
          ae = console.log, ie = console.info, oe = console.warn, se = console.error, ue = console.group, le = console.groupCollapsed, ce = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: fe,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        F++;
      }
    }
    function He() {
      {
        if (F--, F === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: P({}, e, {
              value: ae
            }),
            info: P({}, e, {
              value: ie
            }),
            warn: P({}, e, {
              value: oe
            }),
            error: P({}, e, {
              value: se
            }),
            group: P({}, e, {
              value: ue
            }),
            groupCollapsed: P({}, e, {
              value: le
            }),
            groupEnd: P({}, e, {
              value: ce
            })
          });
        }
        F < 0 && p("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var B = k.ReactCurrentDispatcher, J;
    function Y(e, r, t) {
      {
        if (J === void 0)
          try {
            throw Error();
          } catch (o) {
            var n = o.stack.trim().match(/\n( *(at )?)/);
            J = n && n[1] || "";
          }
        return `
` + J + e;
      }
    }
    var q = !1, L;
    {
      var Be = typeof WeakMap == "function" ? WeakMap : Map;
      L = new Be();
    }
    function de(e, r) {
      if (!e || q)
        return "";
      {
        var t = L.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      q = !0;
      var o = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var s;
      s = B.current, B.current = null, Ue();
      try {
        if (r) {
          var i = function() {
            throw Error();
          };
          if (Object.defineProperty(i.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(i, []);
            } catch (y) {
              n = y;
            }
            Reflect.construct(e, [], i);
          } else {
            try {
              i.call();
            } catch (y) {
              n = y;
            }
            e.call(i.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (y) {
            n = y;
          }
          e();
        }
      } catch (y) {
        if (y && n && typeof y.stack == "string") {
          for (var a = y.stack.split(`
`), m = n.stack.split(`
`), l = a.length - 1, c = m.length - 1; l >= 1 && c >= 0 && a[l] !== m[c]; )
            c--;
          for (; l >= 1 && c >= 0; l--, c--)
            if (a[l] !== m[c]) {
              if (l !== 1 || c !== 1)
                do
                  if (l--, c--, c < 0 || a[l] !== m[c]) {
                    var g = `
` + a[l].replace(" at new ", " at ");
                    return e.displayName && g.includes("<anonymous>") && (g = g.replace("<anonymous>", e.displayName)), typeof e == "function" && L.set(e, g), g;
                  }
                while (l >= 1 && c >= 0);
              break;
            }
        }
      } finally {
        q = !1, B.current = s, He(), Error.prepareStackTrace = o;
      }
      var D = e ? e.displayName || e.name : "", O = D ? Y(D) : "";
      return typeof e == "function" && L.set(e, O), O;
    }
    function Je(e, r, t) {
      return de(e, !1);
    }
    function qe(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function V(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return de(e, qe(e));
      if (typeof e == "string")
        return Y(e);
      switch (e) {
        case j:
          return Y("Suspense");
        case h:
          return Y("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case v:
            return Je(e.render);
          case u:
            return V(e.type, r, t);
          case w: {
            var n = e, o = n._payload, s = n._init;
            try {
              return V(s(o), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var I = Object.prototype.hasOwnProperty, ve = {}, pe = k.ReactDebugCurrentFrame;
    function U(e) {
      if (e) {
        var r = e._owner, t = V(e.type, e._source, r ? r.type : null);
        pe.setExtraStackFrame(t);
      } else
        pe.setExtraStackFrame(null);
    }
    function Ke(e, r, t, n, o) {
      {
        var s = Function.call.bind(I);
        for (var i in e)
          if (s(e, i)) {
            var a = void 0;
            try {
              if (typeof e[i] != "function") {
                var m = Error((n || "React class") + ": " + t + " type `" + i + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[i] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw m.name = "Invariant Violation", m;
              }
              a = e[i](r, i, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (l) {
              a = l;
            }
            a && !(a instanceof Error) && (U(o), p("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, i, typeof a), U(null)), a instanceof Error && !(a.message in ve) && (ve[a.message] = !0, U(o), p("Failed %s type: %s", t, a.message), U(null));
          }
      }
    }
    var Ge = Array.isArray;
    function K(e) {
      return Ge(e);
    }
    function ze(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Xe(e) {
      try {
        return he(e), !1;
      } catch {
        return !0;
      }
    }
    function he(e) {
      return "" + e;
    }
    function me(e) {
      if (Xe(e))
        return p("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ze(e)), he(e);
    }
    var N = k.ReactCurrentOwner, Ze = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, ye, Ee, G;
    G = {};
    function Qe(e) {
      if (I.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function er(e) {
      if (I.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function rr(e, r) {
      if (typeof e.ref == "string" && N.current && r && N.current.stateNode !== r) {
        var t = R(N.current.type);
        G[t] || (p('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', R(N.current.type), e.ref), G[t] = !0);
      }
    }
    function tr(e, r) {
      {
        var t = function() {
          ye || (ye = !0, p("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function nr(e, r) {
      {
        var t = function() {
          Ee || (Ee = !0, p("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var ar = function(e, r, t, n, o, s, i) {
      var a = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: d,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: i,
        // Record the component responsible for creating this element.
        _owner: s
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: o
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function ir(e, r, t, n, o) {
      {
        var s, i = {}, a = null, m = null;
        t !== void 0 && (me(t), a = "" + t), er(r) && (me(r.key), a = "" + r.key), Qe(r) && (m = r.ref, rr(r, o));
        for (s in r)
          I.call(r, s) && !Ze.hasOwnProperty(s) && (i[s] = r[s]);
        if (e && e.defaultProps) {
          var l = e.defaultProps;
          for (s in l)
            i[s] === void 0 && (i[s] = l[s]);
        }
        if (a || m) {
          var c = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && tr(i, c), m && nr(i, c);
        }
        return ar(e, a, m, o, n, N.current, i);
      }
    }
    var z = k.ReactCurrentOwner, ge = k.ReactDebugCurrentFrame;
    function A(e) {
      if (e) {
        var r = e._owner, t = V(e.type, e._source, r ? r.type : null);
        ge.setExtraStackFrame(t);
      } else
        ge.setExtraStackFrame(null);
    }
    var X;
    X = !1;
    function Z(e) {
      return typeof e == "object" && e !== null && e.$$typeof === d;
    }
    function be() {
      {
        if (z.current) {
          var e = R(z.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function or(e) {
      return "";
    }
    var _e = {};
    function sr(e) {
      {
        var r = be();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function Re(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = sr(r);
        if (_e[t])
          return;
        _e[t] = !0;
        var n = "";
        e && e._owner && e._owner !== z.current && (n = " It was passed a child from " + R(e._owner.type) + "."), A(e), p('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), A(null);
      }
    }
    function Te(e, r) {
      {
        if (typeof e != "object")
          return;
        if (K(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            Z(n) && Re(n, r);
          }
        else if (Z(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var o = Fe(e);
          if (typeof o == "function" && o !== e.entries)
            for (var s = o.call(e), i; !(i = s.next()).done; )
              Z(i.value) && Re(i.value, r);
        }
      }
    }
    function ur(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === v || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === u))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = R(r);
          Ke(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !X) {
          X = !0;
          var o = R(r);
          p("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", o || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && p("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function lr(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            A(e), p("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), A(null);
            break;
          }
        }
        e.ref !== null && (A(e), p("Invalid attribute `ref` supplied to `React.Fragment`."), A(null));
      }
    }
    var Se = {};
    function xe(e, r, t, n, o, s) {
      {
        var i = Le(e);
        if (!i) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var m = or();
          m ? a += m : a += be();
          var l;
          e === null ? l = "null" : K(e) ? l = "array" : e !== void 0 && e.$$typeof === d ? (l = "<" + (R(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : l = typeof e, p("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", l, a);
        }
        var c = ir(e, r, t, o, s);
        if (c == null)
          return c;
        if (i) {
          var g = r.children;
          if (g !== void 0)
            if (n)
              if (K(g)) {
                for (var D = 0; D < g.length; D++)
                  Te(g[D], e);
                Object.freeze && Object.freeze(g);
              } else
                p("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Te(g, e);
        }
        if (I.call(r, "key")) {
          var O = R(e), y = Object.keys(r).filter(function(hr) {
            return hr !== "key";
          }), Q = y.length > 0 ? "{key: someKey, " + y.join(": ..., ") + ": ...}" : "{key: someKey}";
          if (!Se[O + Q]) {
            var pr = y.length > 0 ? "{" + y.join(": ..., ") + ": ...}" : "{}";
            p(`A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`, Q, O, pr, O), Se[O + Q] = !0;
          }
        }
        return e === b ? lr(c) : ur(c), c;
      }
    }
    function cr(e, r, t) {
      return xe(e, r, t, !0);
    }
    function fr(e, r, t) {
      return xe(e, r, t, !1);
    }
    var dr = fr, vr = cr;
    $.Fragment = b, $.jsx = dr, $.jsxs = vr;
  }()), $;
}
process.env.NODE_ENV === "production" ? ee.exports = gr() : ee.exports = br();
var f = ee.exports;
const H = {
  SET_THEME: "SET_THEME",
  RESET_THEME: "RESET_THEME"
};
function _r(E, d) {
  switch (d.type) {
    case H.SET_THEME:
      return {
        primary: d.payload.primary,
        secondary: d.payload.secondary
      };
    case H.RESET_THEME:
      return d.payload;
    default:
      return E;
  }
}
const Ae = mr();
function Tr({
  initialTheme: E = { primary: "primary", secondary: "dark" },
  children: d
}) {
  const [C, b] = yr(_r, E), T = {
    setTheme: (x, _) => {
      b({
        type: H.SET_THEME,
        payload: { primary: x, secondary: _ }
      });
    },
    resetTheme: () => {
      b({ type: H.RESET_THEME, payload: E });
    }
  };
  return /* @__PURE__ */ f.jsx(Ae.Provider, { value: { state: C, actions: T }, children: d });
}
function Pe(E) {
  const d = Er(Ae);
  if (d === void 0)
    throw new Error("useTheme must be used within a ThemeProvider");
  return d[E];
}
const Oe = [
  "white",
  "black",
  "light",
  "dark",
  "primary",
  "link",
  "info",
  "success",
  "warning",
  "danger"
];
function Sr() {
  const { primary: E, secondary: d } = Pe("state"), { setTheme: C, resetTheme: b } = Pe("actions"), [T, x] = je(E), [_, S] = je(d), v = (u) => {
    x(u.target.value);
  }, j = (u) => {
    S(u.target.value);
  }, h = (u) => {
    u.preventDefault(), C(T, _);
  };
  return /* @__PURE__ */ f.jsxs("form", { className: "box m-4 p-4", onSubmit: h, children: [
    /* @__PURE__ */ f.jsxs("div", { className: "field", children: [
      /* @__PURE__ */ f.jsx("label", { className: "label", htmlFor: "primary-color", children: "Select Primary Color:" }),
      /* @__PURE__ */ f.jsx("div", { className: "control is-expanded", children: /* @__PURE__ */ f.jsx("div", { className: "select is-fullwidth", children: /* @__PURE__ */ f.jsx(
        "select",
        {
          id: "primary-color",
          value: T,
          onChange: v,
          children: Oe.map((u) => /* @__PURE__ */ f.jsx("option", { value: u, children: u }, u))
        }
      ) }) })
    ] }),
    /* @__PURE__ */ f.jsxs("div", { className: "field", children: [
      /* @__PURE__ */ f.jsx("label", { className: "label", htmlFor: "secondary-color", children: "Select Secondary Color:" }),
      /* @__PURE__ */ f.jsx("div", { className: "control is-expanded", children: /* @__PURE__ */ f.jsx("div", { className: "select is-fullwidth", children: /* @__PURE__ */ f.jsx(
        "select",
        {
          id: "secondary-color",
          value: _,
          onChange: j,
          children: Oe.map((u) => /* @__PURE__ */ f.jsx("option", { value: u, children: u }, u))
        }
      ) }) })
    ] }),
    /* @__PURE__ */ f.jsxs("div", { className: "field is-grouped", children: [
      /* @__PURE__ */ f.jsx("div", { className: "control", children: /* @__PURE__ */ f.jsx("button", { className: "button is-primary", children: "Apply Theme" }) }),
      /* @__PURE__ */ f.jsx("div", { className: "control", children: /* @__PURE__ */ f.jsx(
        "button",
        {
          className: "button is-light",
          type: "button",
          onClick: b,
          children: "Reset Theme"
        }
      ) })
    ] })
  ] });
}
export {
  Ae as ThemeContext,
  Sr as ThemePicker,
  Tr as ThemeProvider,
  Pe as useTheme
};
